vsintegration/src/FSharp.Editor/Navigation/GoToDefinitionService.fs

```diff
@@ -336,7 +336,10 @@ type internal FSharpGoToDefinitionService [<ImportingConstructor>]
                         return results.AsEnumerable()
                     else // we need to get an FSharpSymbol from the targetRange found in the signature
                          // that symbol will be used to find the destination in the corresponding implementation file
-                        let implFilePath = Path.ChangeExtension (sigDocument.FilePath,"fs")
+                        let implFilePath =
+                            // Bugfix: apparently sigDocument not always is a signature file
+                            if isSignatureFile sigDocument.FilePath then Path.ChangeExtension (sigDocument.FilePath, "fs") 
+                            else sigDocument.FilePath
```

Comment:
That's probably not that easy as it seems. I can imagine edge cases, but in a simple case of a script file inside a solution referencing other included scripts and projects I would expect it to just work. There shouldn't be any jumping through hoops to just get symbol info. Project system should handle this somehow, IMO.